app/uk/gov/hmrc/incorporatedentityidentificationfrontend/controllers/CheckYourAnswersController.scala

```diff
@@ -95,7 +95,7 @@ class CheckYourAnswersController @Inject()(journeyService: JourneyService,
             optCtutr <- storageService.retrieveCtutr(journeyId)
             details <- (journeyConfig.businessEntity, optCompanyProfile, optCtutr) match {
               case (LimitedCompany | RegisteredSociety, Some(companyProfile), Some(ctutr)) =>
-                validateIncorporatedEntityDetailsService.validateIncorporatedEntityDetails(companyProfile.companyNumber, ctutr)
+                validateIncorporatedEntityDetailsService.validateIncorporatedEntityDetails(companyProfile.companyNumber, Some(ctutr))
```

Comment:
I think this details block could be simplified a bit, if it's a Limited Company or Registered Society we need to send them to validateIncorporatedEntityDetailsService and if CIO store DetailsNotProvided:
`details <- (journeyConfig.businessEntity, optCompanyProfile) match {
              case (LimitedCompany | RegisteredSociety, Some(companyProfile)) =>
                validateIncorporatedEntityDetailsService.validateIncorporatedEntityDetails(companyProfile.companyNumber, optCtutr)
              case (CharitableIncorporatedOrganisation, Some(_)) => Future.successful(DetailsNotProvided)
              case _ =>
                throw new InternalServerException("Invalid state in Check Your Answers Controller for journeyId: " +  journeyId)
            }`